File: ScalaAlt/src/main/scala/cntrl/Controller.scala
package cntrl

import java.awt.event.{KeyEvent, KeyListener}

import cntrl.keys.{BallRepainter, SideFactory}
import view.View

class Controller extends KeyListener{

  override def keyPressed(keyEvent: KeyEvent): Unit = {
    val repainter: BallRepainter = SideFactory.getMoveSide(keyEvent.getKeyCode)
    repainter.move(View.ball)
  }

  override def keyTyped(keyEvent: KeyEvent): Unit = ???

  override def keyReleased(keyEvent: KeyEvent): Unit = {
    keyPressed(keyEvent)
  }
}

File: KotlinAlter/src/main/kotlin/com/company/cntrl/Controller.kt
package com.company.cntrl

import com.company.cntrl.key.Repainter
import com.company.cntrl.key.SideFactory
import com.company.view.View
import java.awt.event.KeyEvent
import java.awt.event.KeyListener

class Controller : KeyListener {

    override fun keyTyped(keyEvent: KeyEvent?) {}

    override fun keyPressed(keyEvent: KeyEvent?) {
        val repainter: Repainter = SideFactory.getMoveSide(keyEvent!!.keyCode)
        repainter.move(View.ball)
    }

    override fun keyReleased(keyEvent: KeyEvent?) {
        keyPressed(keyEvent)
    }
}